rules.d/90-iocost.rules

```diff
@@ -0,0 +1,20 @@
+#  SPDX-License-Identifier: LGPL-2.1-or-later
+#
+#  This file is part of systemd.
+#
+#  systemd is free software; you can redistribute it and/or modify it
+#  under the terms of the GNU Lesser General Public License as published by
+#  the Free Software Foundation; either version 2.1 of the License, or
+#  (at your option) any later version.
+
+SUBSYSTEM!="block", GOTO="iocost_end"
+
+ACTION=="remove", GOTO="iocost_end"
+
+IMPORT{builtin}="hwdb 'block::name:$env{ID_MODEL}:fwrev:$env{ID_REVISION}:'"
+
+ENV{IOCOST_SOLUTIONS}=="", IMPORT{builtin}="hwdb 'block::name:$env{ID_MODEL}:fwrev::'"
```

Comment:
hmm?

the database can do wildcard matches, you can drop this line without losing anything?

Reply:
There is no issue with the current hwdb file where all pattern ends with fwrev:* (note *)
 There is a concern here when hwdb file has a match pattern without asterisk
e.g block:/dev/sdb:name:*Expansion_HDD*:fwrev:**3489456**:

In this case, if we don't get a match with the first call  we do match based only on the Model name which is done by 2nd call i.e 
`ENV{IOCOST_SOLUTIONS}=="", IMPORT{builtin}="hwdb 'block::name:$env{ID_MODEL}:fwrev::'"
`